extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/filters/accesslog/DefaultAccessLogReceiver.java

```diff
@@ -283,29 +208,21 @@ private void doRotate() {
             if (!Files.exists(defaultLogFile)) {
                 return;
             }
-            Path newFile = outputDirectory.resolve(logBaseName + currentDateString + logNameSuffix);
-            int count = 0;
-            while (Files.exists(newFile)) {
-                ++count;
-                newFile = outputDirectory.resolve(logBaseName + currentDateString + "-" + count + logNameSuffix);
-            }
-            Files.move(defaultLogFile, newFile);
+            Files.move(defaultLogFile,
+                    rotateLogFile(outputDirectory.resolve(logBaseName + currentDateString + logNameSuffix), 0));
         } catch (IOException e) {
             log.error("Error rotating access log", e);
         } finally {
             calculateChangeOverPoint();
         }
     }
 
-    /**
-     * forces a log rotation. This rotation is performed in an async manner, you cannot rely on the rotation
-     * being performed immediately after this method returns.
-     */
-    public void rotate() {
-        forceLogRotation = true;
-        if (stateUpdater.compareAndSet(this, 0, 1)) {
-            logWriteExecutor.execute(this);
+    private Path rotateLogFile(Path newFile, int count) {
+        if (Files.exists(newFile)) {
+            return rotateLogFile(outputDirectory.resolve(logBaseName + currentDateString + "-" + count++ + logNameSuffix),
```

Comment:
having test for double and minus 

<img width="1141" alt="image" src="https://github.com/user-attachments/assets/dbdc1eaa-6ced-4142-a5ff-552045e49d3e" />

but not for missing 


<img width="1151" alt="image" src="https://github.com/user-attachments/assets/e9094765-fc8a-41cc-9099-2b3a5248d58f" />

and pre and before difference: ++count count++